docarray/index/backends/voyager.py

```diff
@@ -0,0 +1,263 @@
+from dataclasses import dataclass, field
+from typing import Any, Dict, Generic, List, Sequence, Tuple, Type, TypeVar, cast
+
+import numpy as np
+from voyager import BaseDoc, BaseDocIndex, DocList, VoyagerBaseDoc
+
+from docarray.utils.find import _FindResult, _FindResultBatched
+
+TSchema = TypeVar('TSchema', bound=VoyagerBaseDoc)
+T = TypeVar('T', bound='VoyagerIndex')
+
+
+@dataclass
+class DBConfig(BaseDocIndex.DBConfig):
+    default_column_config: Dict[Type, Dict[str, Any]] = field(default_factory=dict)
+
+
+@dataclass
+class RuntimeConfig(BaseDocIndex.RuntimeConfig):
+    pass
+
+
+class VoyagerIndex(BaseDocIndex, Generic[TSchema]):
+    def __init__(self, db_config=None, **kwargs):
+        super().__init__(db_config=db_config, **kwargs)
+
+        if not self._db_config or not self._db_config.existing_table:
+            self._create_docs_table()
+
+        self._setup_backend()
+
+    def _create_docs_table(self):
+        columns: List[Tuple[str, str]] = []
+        for col, info in self._column_infos.items():
+            if (
+                col == 'id'
+                or '__' in col
+                or not info.db_type
+                or info.db_type == np.ndarray
+            ):
+                continue
+            columns.append((col, info.db_type))
+
+        columns_str = ', '.join(f'{name} {type}' for name, type in columns)
+        if columns_str:
+            columns_str = ', ' + columns_str
+
+        query = f'CREATE TABLE IF NOT EXISTS docs (doc_id INTEGER PRIMARY KEY, data BLOB{columns_str})'
```

Comment:
does this make that column data will not be filterable at all?

Reply:
The SQL query provided creates a table named `docs` with a `data` column defined as `BLOB` (Binary Large Object). The `BLOB` data type is generally used for storing binary data, and it does not inherently support filtering or indexing.

If the intention is to make the `data` column filterable, I might want to consider using a different data type based on the nature of the data being stored. For example, if the `data` column contains text-based information, changing the data type to `TEXT` could be more appropriate.

Here's an example modification to the query:

```python
if columns_str:
    columns_str = ', ' + columns_str

query = f'CREATE TABLE IF NOT EXISTS docs (doc_id INTEGER PRIMARY KEY, data TEXT{columns_str})'

Reply:
this is what is done in HNSWDocumentIndex I believe

Reply:
Yes like something similar